internal/databases/postgres/backup_push_handler.go

```diff
@@ -21,43 +23,107 @@ import (
 	"github.com/wal-g/wal-g/utility"
 )
 
-type BackupFromFuture struct {
+type sentinelMarshallingError struct {
+	error
+}
+
+func newSentinelMarshallingError(sentinelName string, err error) sentinelMarshallingError {
+	return sentinelMarshallingError{errors.Wrapf(err, "Failed to marshall sentinel file: '%s'", sentinelName)}
+}
+
+func (err sentinelMarshallingError) Error() string {
+	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
+}
+
+type backupFromFuture struct {
 	error
 }
 
-func newBackupFromFuture(backupName string) BackupFromFuture {
-	return BackupFromFuture{errors.Errorf("Finish LSN of backup %v greater than current LSN", backupName)}
+func newBackupFromFuture(backupName string) backupFromFuture {
+	return backupFromFuture{errors.Errorf("Finish LSN of backup %v greater than current LSN", backupName)}
 }
 
-func (err BackupFromFuture) Error() string {
+func (err backupFromFuture) Error() string {
 	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
 }
 
-type BackupFromOtherBD struct {
+type backupFromOtherBD struct {
 	error
 }
 
-func newBackupFromOtherBD() BackupFromOtherBD {
-	return BackupFromOtherBD{errors.Errorf("Current database and database of base backup are not equal.")}
+func newBackupFromOtherBD() backupFromOtherBD {
+	return backupFromOtherBD{errors.Errorf("Current database and database of base backup are not equal.")}
 }
 
-func (err BackupFromOtherBD) Error() string {
+func (err backupFromOtherBD) Error() string {
 	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
 }
 
-type BackupConfig struct {
-	uploader                  *WalUploader
-	archiveDirectory          string
-	backupsFolder             string
-	previousBackupName        string
-	previousBackupSentinelDto BackupSentinelDto
-	isPermanent               bool
-	forceIncremental          bool
-	incrementCount            int
-	verifyPageChecksums       bool
-	storeAllCorruptBlocks     bool
-	tarBallComposerType       TarBallComposerType
-	userData                  string
+// BackupArguments holds all arguments parsed from cmd to this handler class
+type BackupArguments struct {
+	isPermanent           bool
+	verifyPageChecksums   bool
+	storeAllCorruptBlocks bool
+	tarBallComposerType   TarBallComposerType
+	userData              string
+	forceIncremental      bool
+	backupsFolder         string
+	pgDataDirectory       string
+	isFullBackup          bool
+	deltaBaseSelector     internal.BackupSelector
+}
+
+// BackupInfo holds all information that is harvest during the backup process
+type BackupInfo struct {
```

Comment:
There is some duplication between these fields and fields in sentinel/metadata. For example, we have uncompressedSize both in BackupInfo and in BackupInfo.sentinelDto

Also, we don't use this struct in the same way for curBackupInfo and prevBackupInfo.

In prevBackupInfo, we use:
1. sentinel
2. metadata
3. backup name 

Other fields of the prevBackupInfo remain unused.

As for curBackupInfo, we don't actually need to store sentinel and metadata, because they need to be created at the last step, for example in setupDTO(), but we need all the other fields.

So I think it is better to separate previous backup info and current backup info to the separate structs and init the metadata and sentinel in setupDTO() method.


For example, introduce the following structs: 

```
// CurBackupInfo holds all information that is harvest during the backup process
type CurBackupInfo struct {
	name             string
	startTime        time.Time
	startLSN         uint64
	endLSN           uint64
	uncompressedSize int64
	compressedSize   int64
	incrementCount   int
}

// PrevBackupInfo holds all information about previous backup (in case of delta backup)
type PrevBackupInfo struct {
	name             string
	sentinelDto      BackupSentinelDto
	meta             ExtendedMetadataDto
}
```

Reply:
Done